File: ctf-muenster-ui/src/models/Flag.ts
import { Guid } from "guid-typescript";
import Tag from "./Tag";

export default interface Flag {
    id: Guid;
    flagName: string;
    description: string;
    location: {
        latitude: number,
        longitude: number
    };
    dateTimeEndActive: Date;
    dateTimeStartActive: Date;
    tags: Tag[];
}

File: CTFMuensterApi/Model/Flag.cs

using System;
using System.ComponentModel.DataAnnotations;
namespace CTFMuenster.Api.Model
{
    public class Flag {
        [Key]
        public Guid Id {get; set;}
        public String FlagName {get; set;}
        public String Description {get; set;} 
        public Location Location {get; set;} 
        public String ImageFileName {get; set;}
        public ICollection<Tag> Tags {get; set;}
        public DateTimeOffset DateTimeStartActive { get; set; }
        public DateTimeOffset DateTimeEndActive { get; set;}
    }   
}
